src/java/picard/vcf/MendelianViolationDetector.java

```diff
@@ -0,0 +1,338 @@
+    private final List<MendelianViolationMetrics> trios;
+    private final List<Interval> parIntervals;
+    private final double MIN_HET_FRACTION;
+    private final int MIN_GQ;
+    private final int MIN_DP;
+    private final ProgressLogger logger;
+    private final Map<String, Collection<VariantContext>> familyToViolations;
+
+    /** Produces an empty defaulting map that defaults to an empty list of variant contexts.  */
+    private static Map<String, Collection<VariantContext>> makeDefaultingFamilyToViolationsMap() {
+        return new CollectionUtil.DefaultingMap<>(s -> new ArrayList<>(), true);
+    }
+    
+    MendelianViolationDetector(final Set<String> skip_chroms, final Set<String> male_chroms, final Set<String> female_chroms,
+                               final double min_het_fraction, final int min_gq, final int min_dp, final List<MendelianViolationMetrics> trios,
+                               final List<Interval> parIntervals, final ProgressLogger logger) {
+        SKIP_CHROMS = skip_chroms;
+        MALE_CHROMS = male_chroms;
+        FEMALE_CHROMS = female_chroms;
+        MIN_HET_FRACTION = min_het_fraction;
+        MIN_GQ = min_gq;
+        MIN_DP = min_dp;
+        this.trios = trios;
+        this.parIntervals = parIntervals;
+        this.logger = logger;
+        familyToViolations = makeDefaultingFamilyToViolationsMap();
+    }
+
+    /** A little enum to describe the different type of Mendelian violations possible at a bi-allelic site. */
+    enum MendelianViolation {
+        Diploid_Denovo("Parents are both Homozygous Ref and offspring is Heterozygous."),
+        HomVar_HomVar_Het("Parents are both Homozygous Variant and offspring is Heterozygous."),
+        HomRef_HomVar_Hom("One parent is Homozygous Ref, the other Homozygous Variant and the offspring is Homozygous."),
+        Hom_Het_Hom("One parent is Homozygous (Ref or Var), the other is Heterozygous and the offspring is incompatibly Homozygous."),
+        Haploid_Denovo("Offspring variant genotype is expected to be haploid and does not match an expected parental allele."),
+        Haploid_Other("Offspring reference genotype is expected to be haploid and does not match an expected parental allele."),
+        Other("Other unclassified violation of allele transmission.");
+
+        private final String description;
+
+        MendelianViolation(final String desc) {
+            this.description = desc;
+        }
+
+        public String getDescription() {
+            return description;
+        }
+    }
+
+    @Override
+    public void accumulate(final VariantContext ctx) {
+        logger.record(ctx.getContig(), ctx.getStart());
+
+        final String variantChrom = ctx.getContig();
+        final int variantPos = ctx.getStart();
+
+        // Skip anything a little too funky
+        if (ctx.isFiltered()) return;
+        if (!ctx.isVariant()) return;
+        if (!ctx.isSNP()) return;
+        if (!ctx.isBiallelic()) return;
```

Comment:
I think that ideally, it would. but it will take some effort to get the multiallelic code right and I'd rather have something that works for most call-sets than nothing...